### DISCUSSION TITLE
Create releases for old commits?

### DISCUSSION THREAD

**Discussion by GPBeta** [community/community#23535] [New to GitHub]
<p>Hello everyone,</p>
<p>I just want to create tags for some old commits, so I tried to “create a new release” in the releases page of my repository. <img src="https://user-images.githubusercontent.com/13177262/181077769-ace43552-c170-4a77-8995-c1bc979d7859.png" alt="20180420162529.png" data-base62-sha1="gKByNvqicQnRzF7elrapGwvzbrb" width="322" height="330"></p>
<p>However, in Target - Recent Commits drop-down menu, I couldn’t find the commits I wanted to tag with, only very few commits were listed.</p>
<p>So is it possible to create a release for a specified commit on Github?</p>
<p>Thank you for any help :)</p>

**Comment by ThirdEyeRose** [comment, marked as answer]
<p>Hi <a class="mention" href="/gpbeta">@gpbeta</a>,</p>
<p>If you are unable to tag the commit that you want in the GitHub web interface, you can still <a href="https://git-scm.com/book/en/v2/Git-Basics-Tagging" rel="nofollow noopener">tag the commit manually using the Git command line client</a>. Then you can push that tag up to your GitHub repository and make a release based on that tag.</p>
<p>You can also <a href="https://developer.github.com/v3/git/tags/#create-a-tag-object" rel="nofollow noopener">create a tag using the GitHub API</a> as long as you know the SHA of the object that you’re tagging.</p>
<p>Thanks!</p>

**Reply by oezeb** [reply]
Great answer. Worked for me

**Reply by riskeez** [reply]
Check the better answer below [by @Nixinova]( https://github.com/orgs/community/discussions/23535#discussioncomment-3240770)

**Comment by GPBeta** [comment]
<p>Thank you for the solution, it solved my problem :slight_smile:</p>

**Comment by ThirdEyeRose** [comment]
<p>You’re very welcome, <a class="mention" href="/gpbeta">@gpbeta</a>! I’m glad to hear it. :slight_smile:</p>

**Comment by slaughter550** [comment]
<p>Is it on the roadmap to fix this issue, to atleast allow for fuzzy searching? Seems clunky to have to go the API route for a commit more than a few days old.</p>

**Comment by Friduric** [comment]
<p>Looking forward to it!</p>

**Comment by sandvige** [comment]
<p>Hello,</p>
<p>We were also struggling with this limitation, but here is a workarround to manage to create a release on an old commit directly from <a href="http://Github.com" rel="nofollow noopener">Github.com</a> : </p>
<ul>
<li>
<p>Go on the commit direclty (<a href="https://github.com/rgsystemes/rgsupv-core/commit/cddd35f689bea03b7ce3d300f7d57012952861e6" rel="nofollow noopener">https://github.com/xxx/yyy/commit/zzz</a>)</p>
</li>
<li>
<p>Click on the “Browse files” button (<a href="https://github.com/rgsystemes/rgsupv-core/tree/cddd35f689bea03b7ce3d300f7d57012952861e6" rel="nofollow noopener">https://github.com/xxx/yyy/tree/zzz</a>)</p>
</li>
<li>
<p>Create a branch on this commit by hitting the “Tree” drop down, try to find an unexisting branch name, and create it</p>
</li>
<li>
<p>Go on the release creation form, and instead of looking into the “recent commits”, find your freshly created branch</p>
</li>
<li>
<p>Create your release</p>
</li>
<li>
<p>Delete this temporary branch</p>
</li>
</ul>
<p>Enjoy</p>

**Reply by brianjmurrell** [reply]
Of course.  But what a hassle to just create a release from a git hash.

**Comment by noadahlia** [comment]
<p>Simple and fast. Tank you.</p>

**Comment by Nixinova** [comment]
<p>There seems to be support for a “target” URL query. Simply click “new release” and add ?target=&lt;hash&gt; to the URL.</p>

**Reply by jotmaster** [reply]
This should be marked as the answer in my opinion. Thanks!

**Reply by brianjmurrell** [reply]
Well, it could be marked as a work-around.  This really needs support in the existing UI.

**Comment by srl295** [comment]
<p><a class="mention" href="/nixinova">@Nixinova</a> that seemed like a great workaround, <strong>BUT</strong> even though the UI  shows the  from target=, the actual tag created is from the latest, so beware!!</p>

**Reply by bicarlsen** [reply]
I didn't find this to be the case. Using @Nixinova's answer tagged the correct commit.

**Comment by brianjmurrell** [comment]
<p>Yes, this process is terribly cumbersome.  I should be able to put any valid gitref into the <strong>Target</strong> not just a choice of branches and a half-dozen recent commits.  It boggles my mind that even tags cannot be a <strong>Target</strong> value given that GA releases are often made from previous RC tags.  I.e. <em>release</em> 2.0.0, tagged <code>v2.0.0</code> would be made from <code>v2.0.0-rc2</code> once <code>v2.0.0-rc2</code> has been accepted as the GA release.</p>
<p>In fact there should be a specific workflow button on any release to create a new release from it.  I.e. to promote a <code>v2.0.0-rc2</code> <strong>pre-release</strong> to <code>v2.0.0</code> (i.e. final) release.</p>

**Comment by brianjmurrell** [comment]
<p>This should be the accepted solution (well, workaround, really), IMHO.</p>

**Comment by baldwindavid** [comment]
I would love to see this feature fixed to enable folks on our team that are involved with the release, but not with the repo or command line, to be able to specify and document a release. The feature is quite limited in that:
1. Nothing prior to a recent commit can be selected
2. Even for recent commits, a full SHA is not searchable

I would hope that one could copy a full or partial SHA from the commits list page, recent or not, and set it as the target.

**Comment by egvijayanand** [comment]
Allow searching the recent commits with at least the full SHA.

Just missed the threshold by 1 commit and now trying with all other cumbersome workarounds.

Fix this at the earliest. Either load more commits on demand when scrolled to the end or allow searching the commits with SHA.

Regards,
Vijay Anand E G

**Comment by goralczm** [comment]
The easiest solution I found is to change page source code to destinated commit hash.
1. Right click on one of the recent commits and select Inspect
2. Change commit hash to your wanted hash (as in image below)

![Untitled](https://github.com/community/community/assets/95181054/81002072-15c6-479b-a750-790a6152132a)

**Reply by gkurz** [reply]
Worked for me ! Nice hack :wink:

**Reply by mehdibadjian-dfs** [reply]
This definitely still work

**Reply by akira-kurogane** [reply]
Very nice workaround :clap: I was still able to use it today.

**Comment by andybeet** [comment]
Wondering if this is ever going to be addressed formally through GitHub releases page?

**Comment by stevekmcc** [comment]
If there isn't a ridiculous number of commits since the one you want:

1. Start creating a draft release at the oldest commit you can select in Target | Recent Commits
2. Save draft
3. Refresh the page. The list in Target | Recent Commits now has 10 commits with the selected commit first, and going back in time. 
4. If the commit you want isn't in the list yet, select the oldest and repeat steps 2-4 as necessary
5. Select the commit you want and save or publish the release

**Reply by liam-hp** [reply]
This still works

**Reply by teohsinyee** [reply]
works for me

**Comment by broken628781-cell** [comment]
23535

**Comment by broken628781-cell** [comment]
#199286
